packages/flutter/lib/src/widgets/inherited_theme.dart

```diff
@@ -95,27 +99,72 @@ abstract class InheritedTheme extends InheritedWidget {
 
   /// Return a copy of this inherited theme with the specified [child].
   ///
-  /// If the identical inherited theme is already visible from [context] then
-  /// just return the [child].
-  ///
   /// This implementation for [TooltipTheme] is typical:
+  ///
   /// ```dart
   /// Widget wrap(BuildContext context, Widget child) {
-  ///   final TooltipTheme ancestorTheme = context.findAncestorWidgetOfExactType<TooltipTheme>());
-  ///   return identical(this, ancestorTheme) ? child : TooltipTheme(data: data, child: child);
+  ///   return TooltipTheme(data: data, child: child);
   /// }
   /// ```
   Widget wrap(BuildContext context, Widget child);
 
-  /// Returns a widget that will [wrap] child in all of the inherited themes
-  /// which are visible from [context].
-  static Widget captureAll(BuildContext context, Widget child) {
+  /// Returns a widget that will [wrap] `child` in all of the inherited themes
+  /// which are present between `context` and the specified `to`
+  /// [BuildContext].
+  ///
+  /// The `to` context must be an ancestor of `context`. If `to` is not
+  /// specified, all inherited themes up to the root of the widget tree are
+  /// captured.
+  ///
+  /// After calling this method, the themes present between `context` and `to`
+  /// are frozen for the provided `child`. If the themes (or their theme data)
+  /// change in the original subtree, those changes will not be visible to
+  /// the wrapped `child` - unless this method is called again to re-wrap the
+  /// child.
+  static Widget captureAll(BuildContext context, Widget child, {BuildContext? to}) {
     assert(child != null);
     assert(context != null);
 
+    return capture(from: context, to: to).wrap(child);
+  }
+
+  /// Returns a [CapturedThemes] object that includes all the [InheritedTheme]s
+  /// between the given `from` and `to` [BuildContext]s.
+  ///
+  /// The `to` context must be an ancestor of the `from` context. If `to` is
+  /// null, all ancestor inherited themes of `from` up to the root of the
+  /// widget tree are captured.
+  ///
+  /// After calling this method, the themes present between `from` and `to` are
+  /// frozen in the returned [CaptureThemes] object. If the themes (or their
+  /// theme data) change in the original subtree, those changes will not be
+  /// applied to the themes captured in the [CaptureThemes] object - unless this
+  /// method is called again to re-capture the updated themes.
+  ///
+  /// To wrap a [Widget] in the captured themes, call [CapturedThemes.wrap].
+  static CapturedThemes capture({ required BuildContext from, required BuildContext? to }) {
+    assert(from != null);
```

Comment:
This is necessary with NNBD?

Reply:
While our customers still run without NNBD we still need them.